components/patient-data/ui/src/main/resources/PhenoTips/GeneClass.xml

```diff
@@ -98,7 +98,67 @@
     </gene>
     <status>
       <cache>0</cache>
-      <customDisplay/>
+      <customDisplay>{{velocity output=false}}
+#set($sP = $object.xWikiClass.get($name))
+#set ($ALL_VALUES = $sP.listValues)
+#set ($LABEL_MAP = $sP.mapValues)
+#set ($GENE_STATUS_PARENTS = {})
+#set ($GENE_STATUS_FLATLIST = [])
+
+#macro (getFlatCategoriesList)
+    #set ($levels = { "" : 0 })
+    #foreach ($s in $ALL_VALUES)
+      #set ($level = $mathtool.add($levels.get("$!GENE_STATUS_PARENTS.get($s)"), 1))
+      #set ($discard = $levels.put($s, $level))
+      #set ($item = {
+        "id" : $s,
+        "name" : "$!LABEL_MAP.get($s).value",
+        "categoryId" : "$!GENE_STATUS_PARENTS.get($s)",
+        "level" : $level
+      })
+      #set ($discard = $GENE_STATUS_FLATLIST.add($item))
+    #end
+#end
+
+#macro (extractStatusSubtypes)
+    #set ($GENE_STATUS_SUBTYPES = {})
+    #foreach ($v in $ALL_VALUES)
+      #set ($descendents = [])
+      #foreach ($v1 in $ALL_VALUES)
+        #if ($v1.startsWith("${v}_"))
```

Comment:
To circumvent the need for renaming and migrating existing status values, replace the character that marks sub-statuses:
```suggestion
        #if ($v1.startsWith("${v}>"))
```